### PR TITLE
Add Token Verification Feature

### PR DESCRIPTION
**Pull request by BruhLemma-Yadecha**
This pull request introduces a new feature for token verification in the API. It includes testing for both successful and unsuccessful token verification, as well as the implementation of a corresponding view and URL routing. Below are the detailed changes made:

### Changes Made:

- **Created a new view for token verification**:
  - Introduced `TokenVerificationView` in `user_views.py` which handles GET requests to verify tokens.
  - The view checks if the token exists in the database and returns appropriate JSON responses:
    - Returns a 200 status with a message 'Token is valid' for valid tokens.
    - Returns a 401 status with a message 'Token is invalid' for invalid tokens.

- **Added URL routing for the token verification view**:
  - Updated `urls.py` to include a new route `/token/verify/<str:token>/` that maps to the `TokenVerificationView`.

- **Implemented tests for the token verification functionality**:
  - Created a new test class `TokenVerificationTest` in `tests.py` to ensure proper functionality of the token verification view.
    - **Test cases included**:
      - `test_token_verification_success`: Tests valid token verification and checks the success response.
      - `test_token_verification_failure`: Tests response for an invalid token and checks for unauthorized access.

- **Refactored existing components as necessary**:
  - Ensured compatibility and consistency with existing test and view structures while adding the new features.

### CODE REVIEW

**Review by aaladaruncc — APPROVED** [review]
Looks great!

File: api/views/user_views.py [hunk 1]
```diff
@@ -714,3 +714,14 @@ def get(self, request, recording_id):
...
+
+class TokenVerificationView(APIView):
+    permission_classes = [AllowAny]
+
+    def get(self, request, token):
+        if not Token.objects.filter(key=token).exists():
+            return JsonResponse(
+                {"message": "Token is invalid"}, status=status.HTTP_401_UNAUTHORIZED
+            )
+        return JsonResponse({"message": "Token is valid"}, status=status.HTTP_200_OK)
```
Comment (amhajja) [inline api/views/user_views.py:727]: Let's simplify the logic to something like this:
```
class TokenVerificationView(APIView):
    permission_classes = [IsAuthenticated]

    def get(self, request):
        return Response({"message": "Token is valid"}, status=status.HTTP_200_OK)
```

We should also add a `throttle_classes` to this endpoint.

**Review by amhajja — CHANGES_REQUESTED** [review]

File: hice_backend/urls.py [hunk 2]
```diff
@@ -181,4 +181,9 @@
         LectureSummaryView.as_view(),
         name="lecture_summary",
     ),
+    path(
+        "token/verify/<str:token>/",
```
Comment (amhajja) [inline hice_backend/urls.py:185]: Token should not be sent/accepted in URL; it'll be in the authorization header.

**Review by amhajja — CHANGES_REQUESTED** [review]
One last request.

**Review by amhajja — APPROVED** [review]
Looks good; thanks for making the requested changes!

**Pull request merged**